### PR TITLE
Add recipe for immutables 0.5

### PR DESCRIPTION
**Pull request by nicoddemus**

### CODE REVIEW

**Comment by dougalsutherland** [comment]
Our docs aren't up to date on this yet (https://github.com/conda-forge/conda-forge.github.io/pull/576), but I think you want:
```
requirements:
  build:
    - {{ compiler('c') }}
  host:
    - python
    - pip
    - setuptools
  run:
    - python
```

File: recipes/immutables/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+build:
+  skip: True  # [py27]
+  number: 0
+  script: python -m pip install . --no-deps --ignore-installed
+
+requirements:
+  build:
+    - python
+    - pip
+    - setuptools
```
Comment (synapticarbors) [inline recipes/immutables/meta.yaml]: This package looks like it uses c-extensions so please add `toolchain` as a `build` requirement.

File: recipes/immutables/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  source_files:
+    - tests
+  imports:
+    - immutables
+  commands:
+    - pytest tests
+
+about:
+  home: https://github.com/MagicStack/immutables
+  license: Apache Software
```
Comment (dougalsutherland) [inline recipes/immutables/meta.yaml]: `license: Apache-2.0` please.

Comment (synapticarbors) [inline recipes/immutables/meta.yaml]: `license: Apache-2.0`:

https://opensource.org/licenses/alphabetical

File: recipes/immutables/meta.yaml (hunk 1 above)
Comment (dougalsutherland) [reply recipes/immutables/meta.yaml]: Looks like we reviewed at the same time and said the same things. :)

@nicoddemus, adding `toolchain` would also be fine, but https://github.com/conda-forge/staged-recipes/pull/5780#issuecomment-386094742 is the future-proof way that currently has the same effect.

Comment (synapticarbors) [reply recipes/immutables/meta.yaml]: @dougalsutherland, Thanks for the link to the issue. Sometimes it's hard to keep up with best practices/know what's happening in all of the internal discussions on various repos.

Comment (jakirkham) [reply recipes/immutables/meta.yaml]: We know the feeling. :)

If you have any ideas about how we can communicate changes better, please let us know (perhaps as a new issue on the webpage repo).

File: recipes/immutables/meta.yaml (hunk 2 above)
Comment (nicoddemus) [reply recipes/immutables/meta.yaml]: Fixed, thanks!

File: recipes/immutables/meta.yaml (hunk 1 above)
Comment (nicoddemus) [reply recipes/immutables/meta.yaml]: I did not know that, thanks for the pointers. Fixed.

**Comment by nicoddemus** [comment]
For some reason AppVeyor ignored the `skip: True  # [py27]` directive and tried to build on Python 2.7 anyways. Weird, it worked before... any hints folks?

**Comment by nicoddemus** [comment]
Perhaps it was a fluke with AppVeyor

**Comment by ocefpaf** [comment]
> Perhaps it was a fluke with AppVeyor?

Probably. Let's get this in and, if that is really broken, you can fix that in the feedstock. (I saw AppVeyor running even when it should not before.)

**Pull request merged**